### PR TITLE
XS✔ ◾ added microservice boundary rule

### PR DESCRIPTION
**Pull request by lukecookssw**
>
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

PTSD from Microservice architecture

> 2. What was changed?

New rule

> 3. Did you do pair or mob programming (list names)?

NA

### CODE REVIEW

File: rules/review-microservice-boundaries/rule.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+---
+type: rule
+title: Do you regularly review your Microservice boundaries?
+uri: review-microservice-boundaries
+authors:
+  - title: Luke Cook
+    url: https://www.ssw.com.au/people/luke-cook
```
Comment (danielmackay) [inline rules/review-microservice-boundaries/rule.md:7]:
```suggestion
  - title: Luke Cook
    url: https://www.ssw.com.au/people/luke-cook
  - title: Daniel Mackay
    url: https://www.ssw.com.au/people/daniel-mackay
```

File: rules/review-microservice-boundaries/rule.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+* Flag "chatty" services - ie, those services that have the most number of integration points
+* Flag *blocking calls* - ie, if a particular Microservice must wait on downstream/dependent service call before completing
+* Flag *local caches* - ie, if Microservice A keeps a local cache of Microservice B's data
+
+Hotspots are often a **code smell** that could point to Microservices being tightly coupled. These are telltale signs that the boundaries between those services may be in need of adjusting.
+
+## Compare and contrast with BAU
+
+In order to measure your Microservice boundary health, you should see what path a "typical" use case takes through your Microservice.
```
Comment (danielmackay) [inline rules/review-microservice-boundaries/rule.md:35]: Another option here is 'Context Mapping', where by you look at the relationships between different bounded contexts.

**Review by danielmackay — CHANGES_REQUESTED** [review]
Great rule.  Couple of comments.

File: rules/review-microservice-boundaries/rule.md (hunk 2 above)
Comment (lukecookssw) [reply rules/review-microservice-boundaries/rule.md:37]: Done

File: rules/review-microservice-boundaries/rule.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+---
+type: rule
```
Comment (danielmackay) [inline rules/review-microservice-boundaries/rule.md:2]: Needs an `seoDescription`. Can you ChatGPT one up?

**Review by danielmackay — APPROVED** [review]
1 comment

File: rules/review-microservice-boundaries/rule.md (hunk 3 above)
Comment (lukecookssw) [reply rules/review-microservice-boundaries/rule.md:3]: Done

**Pull request merged**